pkg/gui/controllers/helpers/branches_helper.go

```diff
@@ -263,3 +264,34 @@ func (self *BranchesHelper) deleteRemoteBranches(remoteBranches []*models.Remote
 	}
 	return nil
 }
+
+func (self *BranchesHelper) AutoForwardBranches() error {
+	if self.c.UserConfig().Git.AutoForwardBranches == "none" {
+		return nil
+	}
+
+	allBranches := self.c.UserConfig().Git.AutoForwardBranches == "allBranches"
+	branches := self.c.Model().Branches
+	updateCommands := ""
+	// The first branch is the currently checked out branch; skip it
+	for _, branch := range branches[1:] {
```

Comment:
I don't have a use case for auto-forwarding all branches: if you also don't have a use case for it I would just not support it to keep the code simple.

Reply:
The use case is that you collaborate on some branches with a coworker, they push some new commits, and you get them auto-forwarded so that you have those commits right available when you check out the branch again. Saves one pull after checkout. I find this useful and will probably turn it on myself. (The downside is that you don't see that new commits were pushed, so that you can inspect them using "View divergence from upstream").